dvc/render/match.py

```diff
@@ -19,12 +20,34 @@ def get_files(plots_data: Dict) -> List:
     return sorted_files
 
 
+def _get(nested_dict: Dict, path: List[str]):
+    if len(path) < 1:
+        return None
+
+    key = path[0]
+    try:
+        content = nested_dict[key]
+    except KeyError:
+        return None
+
+    if len(path) == 1:
+        return content
+    return _get(content, path[1:])
+
+
 def group_by_filename(plots_data: Dict) -> Dict:
     files = get_files(plots_data)
-    grouped = {
-        file: dpath.util.search(plots_data, ["*", "*", file]) for file in files
-    }
-    return grouped
+    revisions = list(plots_data)
+
+    grouped: Dict[str, Dict] = defaultdict(dict)
+
+    for revision in revisions:
+        for file in files:
+            path = [revision, "data", file]
+            content = _get(plots_data, path)
+            if content:
+                dpath.util.new(grouped[file], path, content)
+    return dict(grouped)
```

Comment:
Assuming that the code above works, why not just

```suggestion
    grouped = {}

    for revision in plots_data.keys():
        data = plots_data[revision].get("data", {})
        for file in data.keys():
            content = data.get(file)
            if content:
                dpath.util.new(grouped, [file, revision, "data", file], content)

    return grouped
```

and not need `get_files`(redundant walk and sort. Btw, not used anywhere, why wasn't it `_get_files`? same question about other functions here),  not need `_get`, and not need revisions (were creating a list for no reason) and reduce complexity?

Reply:
I think it was used somewhere else. Dropping it now.